assets/js/blocks-registry/payment-methods/registry.js

```diff
@@ -67,6 +67,44 @@ export const registerExpressPaymentMethod = ( options ) => {
 	}
 };
 
+/**
+ * Allows extension to register callbacks for specific payment methods to determine if they can make payments
+ *
+ * @param {string} namespace A unique string to identify the extension registering payment method callbacks.
+ * @param {Record<string, function():any>} callbacks Example {stripe: () => {}, cheque: => {}}
+ */
+export const registerPaymentMethodExtensionCallbacks = (
+	namespace,
+	callbacks
+) => {
+	if (
+		Object.values( callbacks ).some(
+			( callback ) => typeof callback !== 'function'
+		)
+	) {
+		const nonFunctionCallbacks = Object.keys( callbacks ).filter(
+			( key ) => typeof callbacks[ key ] !== 'function'
+		);
+
+		throw new Error(
```

Comment:
@opr 👏 for spotting this. 
I am thinking we should not return early, just use `console.error` and proceed to the other callbacks the extension might have. So if one callback is not a function we are registering the ones that actually are. Like so:
```js
export const registerPaymentMethodExtensionCallbacks = (
	namespace,
	callbacks
) => {
	// Set namespace up as an empty object.
	canMakePaymentExtensionsCallbacks[ namespace ] = {};

	Object.entries( callbacks ).forEach(
		( [ paymentMethodName, callback ] ) => {
			if ( typeof callback === 'function' ) {
				canMakePaymentExtensionsCallbacks[ namespace ][
					paymentMethodName
				] = callback;
			} else {
				console.error(
					`All callbacks provided to registerPaymentMethodExtensionCallbacks must be functions. The callback for the ${ paymentMethodName } payment method was not a function. Error occurred when trying to register callbacks for the ${ namespace } namespace.`
				);
			}
		}
	);
};
```

But I do think we should return early if the namespace is already used. So instead of letting a call of registerPaymentMethodExtensionCallbacks that overwrites existing callbacks under the same namespace, just console.error and return. What do you think?

Reply:
> I am thinking we should not return early

OK :)

> But I do think we should return early if the namespace is already used. So instead of letting a call of registerPaymentMethodExtensionCallbacks that overwrites existing callbacks under the same namespace, just console.error and return. What do you think?

Great idea!